pkg/kubectl/cmd/scale.go

```diff
@@ -66,6 +66,9 @@ var (
 func NewCmdScale(f *cmdutil.Factory, out io.Writer) *cobra.Command {
 	options := &ScaleOptions{}
 
+	validArgs := []string{"deployment", "replicaset", "replicationcontroller", "job"}
```

Comment:
@fabianofranz this should probably be hitting the API server to cache discovery for X minutes and navigating the discovered resources for a `scale` subresource, but fall back to this list if it takes too long.  You'll probably want an issue for that.  This one isn't just an openshift problem.